prometheus_hardware_exporter/collector.py

```diff
@@ -832,3 +834,14 @@ def fetch(self) -> List[Payload]:
     def process(self, payloads: List[Payload], datastore: Dict[str, Payload]) -> List[Payload]:
         """Process the payload if needed."""
         return payloads
+
+
+COLLECTOR_REGISTRIES = {
```

Comment:
Not blocker but it would be nice to auto register with annotations
`@collector("mega-raid-collector")`

Reply:
It's not necessary that one class represent one collector, e.g. we have 

```
    "lsi-sas-2-collector": LSISASControllerCollector(2),
    "lsi-sas-3-collector": LSISASControllerCollector(3),
```

Reply:
well, it might work if we do this:

```
COLLECTOR_REGISTRIES= {}

def register(name, *args, **kwargs):
    def decorator(cls):
        COLLECTOR_REGISTRIES[name] = cls(*args, **kwargs)
        return register                                                                                                                     
    return decorator

```

Reply:
I've tried it, and should be working, but the type check effectively make it even more complicated. So if it's not a blocker, then I will try to avoid that.
```
COLLECTOR_REGISTRIES= {}

def register(name, *args, **kwargs):
    def decorator(cls):
        COLLECTOR_REGISTRIES[name] = cls(*args, **kwargs) # type checking in here has no simple solution, because each class has different arguments, and require different types.
        return cls                                                                                                                          
    return decorator

```